common/src/main/java/com/wynntils/utils/type/ErrorOr.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.utils.type;
+
+public final class ErrorOr<T> {
```

Comment:
The class itself is nice, I'm just thinking a bit about the name. What would you think of "ConditionalValue" instead?

Reply:
I think it read fine as `ErrorOr<UsefulType`. Maybe I wrote too much C++... I am fine with renaming, but `ConditionalValue<OtherLongType>` is getting long, isn't it?

Reply:
Ok, fine. Keep ErrorOr.